qml/MainScreen.qml

```diff
@@ -217,6 +227,56 @@ Item {
         defaultValue: "file:///usr/share/asteroid-launcher/applauncher/000-default-horizontal.qml"
     }
 
+    ConfigurationValue {
+        id: watchfaceNightstandSource
+        key: "/desktop/asteroid/nightstand/watchface"
+        defaultValue: "file:///usr/share/asteroid-launcher/watchfaces/005-analog-nordic.qml"
+    }
+
+    ConfigurationValue {
+        id: nightstandBrightness
+        key: "/desktop/asteroid/nightstand/brightness"
+        defaultValue: 30
+    }
+
+    ConfigurationValue {
+        id: nightstandDelay
+        key: "/desktop/asteroid/nightstand/delay"
+        defaultValue: 5
+    }
+
+    ConfigurationValue {
+        id: nightstandEnabled
+        key: "/desktop/asteroid/nightstand/enabled"
+        defaultValue: false
+    }
+
+    Item {
+        id: nightstandMode
+        readonly property bool active: ready || (nightstandEnabled.value && holdoff)
+        readonly property bool ready: nightstandEnabled.value && mceCableState.connected
+        property bool holdoff: false
+        property int oldBrightness: 100
+        onReadyChanged: {
+            if (ready) {
+                oldBrightness = displaySettings.brightness
+                displaySettings.brightness = nightstandBrightness.value
+                holdoff = true
+            } else {
+                displaySettings.brightness = oldBrightness
```

Comment:
Shouldn't the original brightness be restored only when active transitions back to false ? that is when the timer elapses ?

Reply:
There are two circumstances in which it should immediately restore the original brightness:

1. when the timer expires
2. if nightstand mode is disabled

The new version of the code does exactly this.